tests/std/tests/Dev11_0836436_get_time/test.cpp

```diff
@@ -873,3 +873,17 @@ void test_gh_2618() {
     // 1-digit strings: same as 2-digit
     TestTimeGetYear("1", 2001, 1, 2001);
 }
+
+void test_stream_unmatched() {
+
+    const locale loc{locale::classic()};
+    const auto& tmget{use_facet<time_get<char>>(loc)};
+    ios_base::iostate err{ios_base::goodbit};
+    tm when{};
+    const string fmt{"%X"};
+    istringstream iss{"3:04"};
+    istreambuf_iterator<char> first{iss};
+    const istreambuf_iterator<char> last{};

```

Comment:
I don't know exactly, but maybe we need to test with `wchar_t` also, not only `char`